Support Range<NaiveDate> for daterange

the-kenny opened the pull request:
This implements `Normalizable` for `NaiveDate` to support `Range<Date>`. It also changes `assert!` to `assert_eq!` in a few places to improve the test output. 

Note that I had to `#[ignore]` the new test I added because Postgres normalises `(2023-01-01,)` to `[2023-01-02,)`, which makes the test fail as `test_types` does a by-value comparison.

the-kenny committed:
Use `assert_eq!` in `test_type` for better error messages.

the-kenny committed:
with_chrono_0_4: Use NaiveDateTime directly.

This fixes one tests for non-UTC systems.

the-kenny committed:
Test representations of TSRANGE and TZRANGE with time & date

the-kenny committed:
chrono: Add implementations for daterange

the-kenny committed:
chrono: Don't implement Normalizable for NaiveTime

This makes no sense as there is no default 'timerange' type.

sfackler commented on src/impls.rs:
```diff
@@ -173,16 +173,29 @@ mod test {
+
+    #[test]
+    #[cfg(feature = "with-chrono-0_4")]
+    #[ignore = "Exclusive starts will be converted to next-day + inclusive"]
```
I believe this is exactly what the `normalize` method is supposed to handle.

the-kenny replied:
That's good to know, thanks! I'll have another look at it and see if I can adapt my code :-)

the-kenny committed:
with_chrono_0_4: Implement Normalize for NaiveDate correctly.

> The built-in range types int4range, int8range,
> and daterange all use a canonical form that
> includes the lower bound and excludes the upper
> bound; that is, [).

https://www.postgresql.org/docs/current/rangetypes.html#RANGETYPES-DISCRETE

This is implemented by modifying the
`bounded_normalize` macro to take a `$delta`
expression (`1` for `i32` and `i64`,
`Duration::days(1)` for `NaiveDate`).

the-kenny commented:
@sfackler I finally got around to fixing `Normalize` :-) Could I ask you to have another look when your time permits?

lgrosz commented:
Will this get merged?
